compiler/rustc_resolve/src/late.rs

```diff
@@ -1151,23 +1180,59 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                     .span_label(lifetime.ident.span, note)
                     .emit();
 
+                    self.r.lifetimes_res_map.insert(lifetime.id, LifetimeRes::Error);
+                    return;
+                }
+                LifetimeRibKind::AnonymousPassThrough(node_id) => {
+                    self.r
+                        .lifetimes_res_map
+                        .insert(lifetime.id, LifetimeRes::Anonymous { binder: node_id, elided });
                     return;
                 }
-                LifetimeRibKind::AnonymousCreateParameter
-                | LifetimeRibKind::AnonymousPassThrough
-                | LifetimeRibKind::Item => return,
+                LifetimeRibKind::Item => break,
                 _ => {}
             }
         }
+        self.r
+            .lifetimes_res_map
+            .insert(lifetime.id, LifetimeRes::Anonymous { binder: DUMMY_NODE_ID, elided });
     }
 
     #[tracing::instrument(level = "debug", skip(self))]
     fn resolve_elided_lifetime(&mut self, anchor_id: NodeId, span: Span) {
         let id = self.r.next_node_id();
+        self.r.lifetimes_res_map.insert(
+            anchor_id,
+            LifetimeRes::ElidedAnchor { start: id, end: NodeId::from_u32(id.as_u32() + 1) },
+        );
+
         let lt = Lifetime { id, ident: Ident::new(kw::UnderscoreLifetime, span) };
         self.resolve_anonymous_lifetime(&lt, true);
     }
 
+    #[tracing::instrument(level = "debug", skip(self))]
+    fn create_anonymous_lifetime(&mut self, id: NodeId, ident: Ident, item_node_id: NodeId) {
+        debug_assert_eq!(ident.name, kw::UnderscoreLifetime);
+        debug!(?ident.span);
+        let item_def_id = self.r.local_def_id(item_node_id);
+        let def_node_id = self.r.next_node_id();
+        let def_id = self.r.create_def(
+            item_def_id,
+            def_node_id,
+            DefPathData::LifetimeNs(kw::UnderscoreLifetime),
+            ident.span.ctxt().outer_expn(),
```

Comment:
```suggestion
            self.parent_scope.expansion,
```
This should be expansion of the current parent scope, but it's not tracked and does not matter during late resolution, so it can equally well be an `ExpnId::root()` instead, as for definitions created during AST lowering.

Reply:
Same comment applies to other uses of `create_def`.